.gitlab/kernel_version_testing/common.yml

```diff
@@ -0,0 +1,251 @@
+# --- Common scripts
+.shared_filters_and_queries:
+  - FILTER_TEAM="Name=tag:team,Values=ebpf-platform"
+  - FILTER_MANAGED="Name=tag:managed-by,Values=pulumi"
+  - FILTER_STATE="Name=instance-state-name,Values=running"
+  - FILTER_PIPELINE="Name=tag:pipeline-id,Values=${CI_PIPELINE_ID}"
+  - FILTER_ARCH="Name=tag:arch,Values=${ARCH}"
+  - FILTER_INSTANCE_TYPE="Name=tag:instance-type,Values=${INSTANCE_TYPE}"
+  - FILTER_TEST_COMPONENT="Name=tag:test-component,Values=${TEST_COMPONENT}"
+  - QUERY_INSTANCE_IDS='Reservations[*].Instances[*].InstanceId'
+  - QUERY_PRIVATE_IPS='Reservations[*].Instances[*].PrivateIpAddress'
+
+.wait_for_instances:
+  - !reference [.shared_filters_and_queries]
+  - |
+    COUNTER=0
+    while [[ $(aws ec2 describe-instances --filters $FILTER_TEAM $FILTER_MANAGED $FILTER_STATE $FILTER_PIPELINE $FILTER_TEST_COMPONENT --output text --query $QUERY_INSTANCE_IDS  | wc -l ) != "2" && $COUNTER -le 40 ]]; do COUNTER=$[$COUNTER +1]; echo "[${COUNTER}] Waiting for instances"; sleep 30; done
```

Comment:
I just realized this is still waiting for both instances, when it only needs to be waiting for the `ARCH`-specific instance. I think you can add `$FILTER_INSTANCE_TYPE` and change to only expect one result.

Reply:
Maybe rename it to `wait_for_instance` to make that clear too.

Reply:
Done!